manifests/auth/mongodb.pp

```diff
@@ -0,0 +1,117 @@
+# Class: st2::auth::mongodb
+#
+#  Auth class to configure and setup MongoDB Based Authentication
+#
+# Parameters:
+#
+# [*debug*] - Enable Debug (default: false)
+# [*db_host*] - MongoDB Host to connect to (default: 127.0.0.1)
+# [*db_port*] - MongoDB Port to connect to (default: 27017)
+# [*db_name*] - MongoDB DB storing credentials (default: st2auth)
+# [*ssl*] - Enable SSL (default: false)
+# [*ssl_cert*] - Path to SSL Certificate file
+# [*ssl_key*] - Path to SSL Key file
+# [*logging_file*] - Path to logging configuration file
+#
+# Usage:
+#
+#  include ::st2::auth::mongodb
+#
+#  class { 'st2::auth::mongodb':
+#    db_host  => 'mongodb.stackstorm.net',
+#    ssl      => true,
+#    ssl_cert => '/etc/ssl/cert.crt',
+#    ssl_key  => '/etc/ssl/cert.key',
+#  }
+class st2::auth::mongodb (
+  $debug         = false,
+  $db_host       = '127.0.0.1',
+  $db_port       = '27017',
+  $db_name       = 'st2auth',
+  $ssl           = false,
+  $ssl_cert      = undef,
+  $ssl_key       = undef,
+  $logging_file  = '/etc/st2api/logging.conf',
+) {
+  $_debug = $debug ? {
+    true    => 'True',
+    default => 'False',
+  }
+  $_ssl = $ssl ? {
+    true    => 'True',
+    default => 'False',
+  }
+  $_api_url = $::st2::api_url
+
+  ini_setting { 'auth_mode':
+    ensure  => present,
+    path    => '/etc/st2/st2.conf',
+    section => 'auth',
+    setting => 'mode',
+    value   => 'standalone',
+  }
+  ini_setting { 'auth_backend':
+    ensure  => present,
+    path    => '/etc/st2/st2.conf',
+    section => 'auth',
+    setting => 'backend',
+    value   => 'mongodb',
+  }
+  ini_setting { 'auth_backend_kwargs':
+    ensure  => present,
+    path    => '/etc/st2/st2.conf',
+    section => 'auth',
+    setting => 'backend_kwargs',
+    value   => "{\"db_host\": \"${db_host}\", \"db_port\": \"${db_port}\", \"db_name\": \"${db_name}\"}",
```

Comment:
Can you use some kind if "json.stringify" here? Or is this not possible since you can't include arbitrary Ruby code in the manifests?